### PR TITLE
Fix MVFCheckOverflowCalculation_test. Set mvf-bu-retarget.py to 90 days.

### PR DESCRIPTION
**Pull request by marlengit**
Some rework to tidy up a few bumps in the testing:
* reduced fork retarget period to 3 months (this makes testing faster later revert to 6 months if necessary)
* Added "-forkheight=999999" option to the  p2p-versionbits-warning.py test.
* MVFCheckOverflowCalculation_test - replaced a hard code

### CODE REVIEW

**Comment by ftrader** [comment]
>reduced fork retarget period to 3 months

If you mean to do this for the whole client instead of just the Python test, then I think you need to adapt src/mvf-bu.h, this line:

```HARDFORK_RETARGET_BLOCKS = 25920,    // 180*144 blocks```

If you only meant to shorten the test duration, then please add an `MVF-BU TODO:` comment next to the change in the Python script which you want to revert, so we can find this easily later.

Please also remove test_framework/arith.py (added by 	e732d70) unless you're using some of it in the tests (I don't see any commits which do, which makes me presume you accidentally committed it).

File: qa/rpc-tests/mvf-bu-retarget.py [hunk 1]
```diff
@@ -15,7 +15,7 @@
 import decimal
 
 # period (in blocks) from fork activation until retargeting returns to normal
-HARDFORK_RETARGET_BLOCKS = 180*144
+HARDFORK_RETARGET_BLOCKS = 90*144
```
Comment (ftrader) [inline qa/rpc-tests/mvf-bu-retarget.py:19]: please add `MVF-BU TODO:` comment here to remind us that we have to revert this later

File: src/test/pow_tests.cpp [hunk 2]
```diff
@@ -122,7 +124,7 @@ BOOST_AUTO_TEST_CASE(MVFCheckOverflowCalculation_test)
     // need to set -force-retarget, otherwise cannot test overflow
     // because it would never reach the computation
     SoftSetBoolArg("-force-retarget", true);
-    BOOST_CHECK_EQUAL(CalculateNextWorkRequired(&pindexLast, nLastRetargetTime, params), 0x207fffff);
+    BOOST_CHECK_EQUAL(CalculateNextWorkRequired(&pindexLast, nLastRetargetTime, params), bnPowLimit.GetCompact());
```
Comment (ftrader) [inline src/test/pow_tests.cpp:127]: Nice, I had thought about making this kind of change too.

File: qa/rpc-tests/mvf-bu-retarget.py (hunk 1 above)
Comment (marlengit) [reply qa/rpc-tests/mvf-bu-retarget.py:19]: Done but it will come through with the first req7/forkresetwork commits shortly.

**Comment by ftrader** [comment]
If you plan on making use of the code which you've checked in as arith.py, we need to go over that file in detail and establish which parts are licensed under what licenses.
Also, it needs cleanup, a file header, etc. It would be best to remove functions from it that are not needed.

We should discuss this on Slack in more detail.
In the meantime, I suggest to remove it from this PR if it is not used by tests here.
